From 2015092ce9a2c2618651a3aca6e1672c404fd27b Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Fri, 25 Jan 2013 11:04:43 +0800
Subject: [PATCH 410/502] ENGR00241722 Remove blank ioctl to HDMI fb dev

Need to keep HDMI pixel clock in early suspend state
to make audio be able to output through DHMI

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx6/hwcomposer/hwcomposer.cpp | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/mx6/hwcomposer/hwcomposer.cpp b/mx6/hwcomposer/hwcomposer.cpp
index 94e7553..50ed2f1 100755
--- a/mx6/hwcomposer/hwcomposer.cpp
+++ b/mx6/hwcomposer/hwcomposer.cpp
@@ -235,15 +235,19 @@ static int hwc_blank(struct hwc_composer_device_1 *dev, int disp, int blank)
         return 0;
     }
 
-    if(ctx->m_viv_hwc) {
+    if (ctx->m_viv_hwc) {
         ctx->m_viv_hwc->blank(ctx->m_viv_hwc, disp, blank);
     }
 
-    int fb_blank = blank ? FB_BLANK_POWERDOWN : FB_BLANK_UNBLANK;
-    int err = ioctl(ctx->mDispInfo[disp].fd, FBIOBLANK, fb_blank);
-    if (err < 0) {
-        ALOGE("blank ioctl failed");
-        return -errno;
+    //HDMI need to keep unblank since audio need to be able to output
+    //through HDMI cable. Blank the HDMI will lost the HDMI clock
+    if (ctx->mDispInfo[disp].type !=  HWC_DISPLAY_HDMI) {
+        int fb_blank = blank ? FB_BLANK_POWERDOWN : FB_BLANK_UNBLANK;
+        int err = ioctl(ctx->mDispInfo[disp].fd, FBIOBLANK, fb_blank);
+        if (err < 0) {
+            ALOGE("blank ioctl failed");
+            return -errno;
+        }
     }
 
     return 0;
-- 
1.8.0

